lib/thrift/servers/binary_framed/acceptor_supervisor.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Thrift.Servers.BinaryFramed.AcceptorSupervisor do
+  @moduledoc """
+  A supervisor for acceptor processes.
+  """
+
+  alias Thrift.Servers.BinaryFramed.Acceptor
+  use Supervisor
+
+  def start_link(socket, server_module, handler_module, count) do
+    Supervisor.start_link(__MODULE__, [socket, server_module, handler_module, count])
+  end
+
+  def init([socket, server_module, handler_module, count]) do
+    children = for id <- (1..count) do
+      worker_id = "#{inspect handler_module}_acceptor_#{id}"
+      worker(Acceptor, [socket, server_module, handler_module], id: worker_id)
+    end
+    supervise(children, strategy: :one_for_one)
```

Comment:
This is a really solid point. I'm still pretty terrible with supervision; and had a similar problem a while back. 
I don't know if we need _telecom grade_ fault tolerance in this server, but it should be tolerant of losing all its clients, as many of our clients only hold connections for 5 minutes.